### PR TITLE
PT/Glow operator SoftMax.

### PR DESCRIPTION
**Pull request by putivsky**
Summary:
PyTorch - Glow integration for SoftMax operator.
Documentation:

[Optional Fixes #issue]

Test Plan:

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: torch_glow/tests/nodes/softmax_test.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import torch
+import torch.nn.functional as F
+import torch_glow
+
+from tests.utils import jitVsGlow
+
+# Basic test of the PyTorch SoftMax Node on Glow.
+
+
+def test_softmax_basic():
```
Comment (jackm321) [inline torch_glow/tests/nodes/softmax_test.py]: Use the format
```
def test_softmax_basic():
    """Basic test of the PyTorch SoftMax Node on Glow."""
    ...
```

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -734,6 +737,100 @@ llvm::Error PyTorchModelLoader::loadMin(const torch::jit::Node *ptNode) {
   return addGlowNodeValue(outputs[0], output);
 }
 
+llvm::Error PyTorchModelLoader::loadTopK(const torch::jit::Node *ptNode) {
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: Since we can't fuse topK (right?) maybe we should wait on this since we can't test it yet.

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 3]
```diff
@@ -734,6 +736,51 @@ llvm::Error PyTorchModelLoader::loadMin(const torch::jit::Node *ptNode) {
...
+    RETURN_IF_ERR(contractParamIfNeeded(dimHandle, dim));
+  }
+
+  // Dtype (optional).
+  glow::unsigned_t dtype = 1;
+  if (hasGlowNodeValue(inputs[Inputs::dtype])) {
+    auto dtypeHandle = glow::Handle<int32_t>::createInvalidHandle();
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        dtypeHandle, getGlowConstantHandle<int32_t>(inputs[Inputs::dtype]));
+    RETURN_IF_ERR(contractParamIfNeeded(dtypeHandle, dtype));
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: don't need contractParamIfNeeded

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 4]
```diff
@@ -734,6 +736,51 @@ llvm::Error PyTorchModelLoader::loadMin(const torch::jit::Node *ptNode) {
...
+  glow::unsigned_t dim = 1;
+  if (hasGlowNodeValue(inputs[Inputs::dim])) {
+    auto dimHandle = glow::Handle<int32_t>::createInvalidHandle();
+    ASSIGN_VALUE_OR_RETURN_ERR(
+        dimHandle, getGlowConstantHandle<int32_t>(inputs[Inputs::dim]));
+    RETURN_IF_ERR(contractParamIfNeeded(dimHandle, dim));
+  }
+
+  // Dtype (optional).
+  glow::unsigned_t dtype = 1;
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: This isn't used anywhere and also should use something other than 1 like an enum from somewhere if available not sure where torch.dtype can be found (maybe ScalarType?)

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 5]
```diff
@@ -734,6 +736,51 @@ llvm::Error PyTorchModelLoader::loadMin(const torch::jit::Node *ptNode) {
...
+      dtype = 2,
+    };
+  };
+
+  glow::NodeValue in;
+  ASSIGN_VALUE_OR_RETURN_ERR(in, getGlowNodeValue(inputs[Inputs::input]));
+
+  // Dim.
+  glow::unsigned_t dim = 1;
+  if (hasGlowNodeValue(inputs[Inputs::dim])) {
```
Comment (jackm321) [inline torch_glow/src/PyTorchModelLoader.cpp]: this is not an optional input

File: torch_glow/tests/nodes/softmax_test.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import torch
+import torch.nn.functional as F
+import torch_glow
```
Comment (jackm321) [inline torch_glow/tests/nodes/softmax_test.py]: not used

File: torch_glow/tests/nodes/softmax_test.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+from tests.utils import jitVsGlow
+
+
+def test_softmax_basic():
+    def softmax_basic(inputs):
+        """Basic test of the PyTorch SoftMax Node on Glow."""
```
Comment (jackm321) [inline torch_glow/tests/nodes/softmax_test.py]: move to top level function
